### PR TITLE
1.0.4 patch train updates

### PR DESCRIPTION
**Pull request by JunTaoLuo**

### CODE REVIEW

File: test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs [hunk 1]
```diff
@@ -18,7 +19,7 @@ public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File_Explicitly_Passed(
                                 .AddUserSecrets(userSecretsId: Guid.NewGuid().ToString());
         }
 
-        [Fact]
+        [ConditionalFact (Skip = "Obsolete API")]
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs]: You don't need ConditionalFact to do this. `[Fact(Skip = "reason")]`.

Also, don't skip these tests. They should still work even though API is obsolete.

File: test/Microsoft.Extensions.SecretManager.Tests/project.json [hunk 2]
```diff
@@ -1,11 +1,15 @@
 {
   "buildOptions": {
     "warningsAsErrors": true,
+    "nowarn": [
+      "CS0618" // Tests using obsolete APIs are skipped.
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.SecretManager.Tests/project.json]: Don't add a project-wide nowarn. Use `#pragma` to disable it. Example: 
image

File: test/Microsoft.Extensions.SecretManager.Tests/project.json [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+    ],
     "keyFile": "../../tools/Key.snk"
   },
   "dependencies": {
     "dotnet-test-xunit": "1.0.0-rc3-000000-01",
-    "Microsoft.Extensions.SecretManager.Tools": "1.0.0-preview4-final",
+    "Microsoft.AspNetCore.Testing": "1.0.1",
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.SecretManager.Tests/project.json]: We don't need to add a new dependency. ConditionalFacts not required.

File: test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs (hunk 1 above)
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs]: Ahh

Comment (JunTaoLuo) [inline test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs]: Actually I don't think there is much value keeping these tests around. Why not just remove them?

File: test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs [hunk 4]
```diff
@@ -19,37 +19,6 @@ public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File_Explicitly_Passed(
         }
 
         [Fact]
-        public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File()
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.SecretManager.Tests/ConfigurationExtensionTests.cs]: Please keep these. Even though we've obsoleted the API, we want to ensure we haven't regressed on the tool.

**Comment by JunTaoLuo** [comment]
No longer building this repo in 1.0.4 patch train.

**Pull request closed without merging**